sqlmesh/core/config/connection.py

```diff
@@ -84,15 +85,21 @@ def is_recommended_for_state_sync(self) -> bool:
         """Whether this connection is recommended for being used as a state sync for production state syncs"""
         return self.type_ in RECOMMENDED_STATE_SYNC_ENGINES
 
+    @property
+    def _connection_factory_with_kwargs(self) -> t.Callable:
```

Comment:
Should we more specific with type here? Eg. specify that it expects no arguments